.github/ci/update_supported_boards.py

```diff
@@ -0,0 +1,235 @@
+                        if mic_code_section is not None:
+                            matches = re.findall(r"([a-z0-9_]+)_codec_new", mic_code_section)
+                            mic_codec = [m for m in matches if not m.startswith("audio")]
+
+    # Extract SoC from YML
+    with open(yml_file, "r", encoding="utf-8") as f:
+        yml_data = yaml.safe_load(f)
+        soc = yml_data.get("targets", ["-"])[0]
+
+    # Prepare image path
+    image_path = next(board_path.glob(f"doc/{board_name}.webp"), None)
+    img_html = f'<img src="{image_path.as_posix()}" width="150">' if image_path.exists() else "-"
+
+    # Emoji map for features
+    feature_emojis = {
+        "DISPLAY": ":pager:",
+        "TOUCH": ":point_up:",
+        "AUDIO": ":musical_note:",
+        "AUDIO_MIC": ":microphone:",
+        "AUDIO_SPEAKER": ":speaker:",
+        "SDCARD": ":floppy_disk:",
+        "IMU": ":video_game:",
+        "USB": ":electric_plug:",
+        "BUTTONS": ":radio_button:",
+        "LED": ":bulb:",
+        "CAMERA": ":camera:",
+        "BAT": ":battery:",
+        "KNOB": ":white_circle:",
+    }
+
+    # Name map for features
+    feature_names = {
+        "DISPLAY": "LCD Display",
+        "TOUCH": "Display Touch",
+        "AUDIO": "Audio",
+        "AUDIO_MIC": "Audio Microphone",
+        "AUDIO_SPEAKER": "Audio Speaker",
+        "SDCARD": "uSD Card",
+        "IMU": "IMU",
+        "USB": "USB-OTG",
+        "BUTTONS": "Button",
+        "LED": "LED",
+        "CAMERA": "Camera",
+        "BAT": "Battery",
+        "KNOB": "Knob",
+    }
+
+    feature_list = ""
+    for f in sorted(features):
+        f_emoji = feature_emojis.get(f, ":black_circle:")
+        f_name = feature_names.get(f, f)
+        f_suffix = ""
+
+        if f == "DISPLAY":
+            f_suffix = f" ({', '.join(display_types)})" if display_types else ""
+        elif f == "TOUCH":
+            f_suffix = f" ({', '.join(touch_types)})" if touch_types else ""
+        elif f == "AUDIO_SPEAKER":
+            f_suffix = f" ({', '.join(speaker_codec)})" if speaker_codec else ""
+        elif f == "AUDIO_MIC":
+            f_suffix = f" ({', '.join(mic_codec)})" if mic_codec else ""
+
+        feature_list += f"{f_emoji} {f_name} {f_suffix}<br/>"
+    feature_list += ""
+
+    return {
+        "name": bsp_name,
+        "link": f"[{bsp_name}](bsp/{board_name})",
+        "soc": soc,
+        "features": f"{feature_list}",
+        "photo": img_html,
+        "deprecated": deprecated,
+    }
+
+
+def generate_table():
+    header = "| Board name | SoC | Supported Features | Photo |\n|:----------:|:---:|:-------------------|:-----:|"
+    active_rows = []
+    deprecated_rows = []
+
+    for board_path in sorted(BSP_DIR.iterdir()):
+        if board_path.is_dir():
+            info = extract_board_info(board_path)
+            if info:
+                row = f"| {info['link']} | {info['soc']} | {info['features']} | {info['photo']} |"
```

Comment:
Also just an FYI, not need to update:

I used `from py_markdown_table.markdown_table import markdown_table` in [update_dependencies.py](https://github.com/espressif/esp-bsp/blob/master/.github/ci/update_readme_dependencies.py#L16), this will give you more flexibility in table formatting compared to this 'raw' table writting

Reply:
I know about it. I tried it, but there is not possible to use for example aligment like this `| :---: |`  and generated table sometimes looks bad in code :-(